lib/src/conflicts.rs

```diff
@@ -77,51 +79,75 @@ fn write_diff_hunks(hunks: &[DiffHunk], file: &mut dyn Write) -> std::io::Result
     Ok(())
 }
 
-async fn get_file_contents(store: &Store, path: &RepoPath, term: &Option<FileId>) -> ContentHunk {
+#[derive(Error, Debug)]
+pub enum MaterializationError {
+    #[error(transparent)]
+    Backend(#[from] BackendError),
+    #[error("Failed to read file contents for {} with id {}: {source}", path.as_internal_file_string(), id.hex())]
+    Read {
+        path: RepoPathBuf,
+        id: FileId,
+        source: std::io::Error,
+    },
+    #[error(transparent)]
+    Write(std::io::Error),
+}
+
+async fn get_file_contents(
+    store: &Store,
+    path: &RepoPath,
+    term: &Option<FileId>,
+) -> Result<ContentHunk, MaterializationError> {
     match term {
         Some(id) => {
             let mut content = vec![];
             store
                 .read_file_async(path, id)
-                .await
-                .unwrap()
+                .await?
                 .read_to_end(&mut content)
-                .unwrap();
-            ContentHunk(content)
+                .map_err(|err| MaterializationError::Read {
+                    path: path.to_owned(),
+                    id: id.clone(),
+                    source: err,
+                })?;
```

Comment:
Perhaps, `MaterializationError::Read` can be merged with `BackendError::ReadObject`? `io::Error` here basically means backend failed to read blob. It's unclear which "read error" should be returned.

This will also simplify error propagation to `SnapshotError`, which has two `BackendError` slots (`BackendError(_)` and `MaterializationError(Backend(_))`).

Reply:
I suppose I could create a new `BackendError::ReadContent`. I'm not sure how I feel about having higher-level code produce `BackendError`s, but I agree that it's convenient. I think I've created `io::Error`s in the past and it was convenient but seemed ugly. Would you prefer such a `BackendError::ReadContent` variant?

Reply:
Done. That turned out a lot simpler. Thanks.